internal/clients/vault_secrets_preview.go

```diff
@@ -61,27 +61,34 @@ func OpenVaultSecretsAppSecrets(ctx context.Context, client *Client, loc *shared
 
 	var secrets *secret_service.OpenAppSecretsOK
 	var err error
-	for attempt := 0; attempt < retryCount; attempt++ {
-		secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
-		if err != nil {
-			var serviceErr *secret_service.OpenAppSecretDefault
-			ok := errors.As(err, &serviceErr)
-			if !ok {
+	var result []*secretmodels.Secrets20231128OpenSecret
+
+	for {
+		for attempt := 0; attempt < retryCount; attempt++ {
+			secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
+			if err != nil {
+				var serviceErr *secret_service.OpenAppSecretDefault
+				ok := errors.As(err, &serviceErr)
+				if !ok {
+					return nil, err
+				}
+				if shouldRetryWithSleep(ctx, serviceErr, attempt, []int{http.StatusTooManyRequests}) {
+					continue
+				}
 				return nil, err
 			}
-			if shouldRetryWithSleep(ctx, serviceErr, attempt, []int{http.StatusTooManyRequests}) {
-				continue
+			if secrets == nil {
+				return nil, errors.New("unable to get secrets")
 			}
-			return nil, err
+			result = append(result, secrets.GetPayload().Secrets...)
```

Comment:
Just curious, could `secrets.GetPayload()` return nil if `secrets` is not nil?

Reply:
No idea but I don't think it is a problem because this code returned secrets.GetPayload().Secrets even before my change for pagination support and we never had issues before. 
I looked at other usages of GetPayload() in this repo and did not find any checks for if GetPayload() is nil, so I don't think it can be nil if `secrets` is not nil.

Reply:
No idea but I don't think it is a problem because this code returned secrets.GetPayload().Secrets even before my change for pagination support and we never had issues before. 
I looked at other usages of GetPayload() in this repo and did not find any checks for if GetPayload() is nil, so I don't think it can be nil if `secrets` is not nil.

Reply:
No idea but I don't think it is a problem because this code returned secrets.GetPayload().Secrets even before my change for pagination support and we never had issues before. 
I looked at other usages of GetPayload() in this repo and did not find any checks for if GetPayload() is nil, so I don't think it can be nil if `secrets` is not nil.